test/flows/calculate_statutory_sick_pay_test.rb

```diff
@@ -0,0 +1,598 @@
+      should "not allow sick day to be before 2010" do
+        assert_invalid_response "2009-01-01"
+      end
+
+      should "not allow a sick day in the future" do
+        assert_invalid_response "3000-01-01"
+      end
+
+      should "not allow the linked sickness to end more than 8 weeks before first sick day" do
+        assert_invalid_response "2019-10-01"
+      end
+
+      should "make sure the linked sickness is before first sick day" do
+        assert_invalid_response "2020-01-01"
+      end
+
+      should "make sure linked sickness ends after it begins" do
+        assert_invalid_response "2019-11-30"
+      end
+    end
+  end
+
+  context "paid_at_least_8_weeks" do
+    setup do
+      testing_node :paid_at_least_8_weeks?
+      add_responses is_your_employee_getting?: "statutory_adoption_pay",
+                    coronavirus_related?: "no",
+                    employee_tell_within_limit?: "yes",
+                    employee_work_different_days?: "no",
+                    first_sick_day?: "2020-01-01",
+                    last_sick_day?: "2020-01-05",
+                    has_linked_sickness?: "yes",
+                    linked_sickness_start_date?: "2019-12-01",
+                    linked_sickness_end_date?: "2019-12-30"
+    end
+
+    context "next_node" do
+      should "have a next node of how_often_pay_employee_pay_patterns for 'eight_weeks_more' response" do
+        assert_next_node :how_often_pay_employee_pay_patterns?, for_response: "eight_weeks_more"
+      end
+
+      should "have a next node of total_earnings_before_sick_period for 'eight_weeks_less' response" do
+        assert_next_node :total_earnings_before_sick_period?, for_response: "eight_weeks_less"
+      end
+
+      should "have a next node of how_often_pay_employee_pay_patterns for 'before_payday' response" do
+        assert_next_node :how_often_pay_employee_pay_patterns?, for_response: "before_payday"
+      end
+    end
+  end
+
+  context "how_often_pay_employee_pay_patterns" do
+    setup do
+      testing_node :how_often_pay_employee_pay_patterns?
+      add_responses is_your_employee_getting?: "statutory_adoption_pay",
+                    coronavirus_related?: "no",
+                    employee_tell_within_limit?: "yes",
+                    employee_work_different_days?: "no",
+                    first_sick_day?: "2020-01-01",
+                    last_sick_day?: "2020-01-05",
+                    has_linked_sickness?: "yes",
+                    linked_sickness_start_date?: "2019-12-01",
+                    linked_sickness_end_date?: "2019-12-30"
+    end
+
+    context "next_node" do
+      should "have a next node of last_payday_before_sickness when paid at least 8 weeks of earnings" do
+        add_responses paid_at_least_8_weeks?: "eight_weeks_more"
+        assert_next_node :last_payday_before_sickness?, for_response: "weekly"
+      end
+
+      should "have a next node of last_payday_before_sickness when before payday" do
```

Comment:
It's not just for "before_payday", so how about:

```suggestion
      should "have a next node of last_payday_before_sickness when not paid 8 weeks of earnings" do
```

Reply:
Good shout, have fixed